### ISSUE TITLE
how do I create a minecraft mod with this?

### ISSUE THREAD

**Issue by marlester-dev** [CrackedPolishedBlackstoneBricksMC/voldeloom#2]
There is no normal guide for this. I want to create a mc 1.6.4 mod with it, how do I do it? Is it fabric-loom for forge? Does it replace forgegradle?

**Comment by marlester-dev** [comment]
found something

**Comment by quat1024** [comment]
Dunno why github didn't notify me for this comment, sorry! I really do need to put some ready-to-use sample projects somewhere (the `./sample` folder in this repo is for testing only)

> Is it fabric-loom for Forge?

It is *somewhat* based off the code in a *very old version* of Fabric Loom. But the projects have diverged significantly since the fork point wayyy back in 2020, and I don't think the core of Voldeloom has many similarities to Fabric Loom anymore. All code relating to the Fabric modloader has been removed. So it's kind of its own thing.

> Does it replace forgegradle?

It "replaces ForgeGradle" in spirit - if you want to set up a mod for an older version of the game, you can use this *instead* of ForgeGradle. But it is not *compatible* with ForgeGradle. If you have an existing ForgeGradle project, changing it to use Voldeloom is more complicated than just swapping the plugin. So it's not a "replacement" in that sense.

Here's a complete `build.gradle` template for a 1.6.4 mod. Use Gradle 7.

```gradle
buildscript {
	repositories {
		mavenCentral()
		maven { url "https://maven.fabricmc.net" }
		maven { url "https://repo.sleeping.town" }
	}
	dependencies {
		classpath "agency.highlysuspect:voldeloom:2.3"
	}
}

apply plugin: "agency.highlysuspect.voldeloom"

java.toolchain.languageVersion = JavaLanguageVersion.of(11)
compileJava.options.release.set(6)

String minecraftVersion = "1.6.4"
String forgeVersion = "1.6.4-9.11.1.1345"

dependencies {
	minecraft "com.mojang:minecraft:${minecraftVersion}"
	forge "net.minecraftforge:forge:${forgeVersion}:universal"
	mappings "net.minecraftforge:forge:${forgeVersion}:src@zip"
}
```

**Comment by Consumedgrub2** [comment]
So for clarity, since I need to do the same thing as OP, (Im new to ancient minecraft modding) this does make a forge mod that can be loaded with fml? Right? (Also kinda new to modding, just want to attempt to port an old childhood mod) (Apologies, extremely dumb question but I just see fabric in the in some places of the project so I just want to be sure but I wanna say yes to my question) (I only know about forge not fabric, not really familiar with fabric)

**Comment by exaskye** [comment]
Yes, Voldeloom emits standard Forge/FML mods loadable in a completely standard Forge environment. No tricks.

The codebase is originally based on Fabric Loom, but has heavily diverged at this point. It's a genetic relation, not a functional one.
